content/docs/nextjs/github-public-repo.md

```diff
@@ -0,0 +1,291 @@
+- You must be using **Next.js v.9.3** or later.
+- If your site is static, is must be hosted on a service that supports Next.js preview-mode (E.g [Zeit's Now](https://zeit.co/home))
+- Your editable content must be stored in GitHub
+- Your repository containing your content **must be public**
+
+## Installation
+
+We'll need to add a few Tina-GitHub packages to our site:
+
+```
+npm install --save react-tinacms-github next-tinacms-github
+```
+
+or
+
+```
+yarn add react-tinacms-github next-tinacms-github
+```
+
+#### `react-tinacms-github`
+
+This package provides helpers for setting up TinaCMS to use the GitHub API, with GitHub authentication.
+
+#### `next-tinacms-github`
+
+This package provides helpers for managing the github auth token for requests, as well as
+providing helpers for loading content from the GitHub API.
+
+### Environment Variables
+
+Throughout this guide, some environment variables will be used:
+
+You may want a `.env` file in your project root i.e
+
+```
+# .env
+GITHUB_CLIENT_ID= # We will generate this later when we create a GitHub app.
+GITHUB_CLIENT_SECRET= # We will generate this later when we create a GitHub app.
+REPO_FULL_NAME=tinacms/tinacms.org # This is your github repository's owner / repo-name.
+BASE_BRANCH=master
+```
+
+To load these .env values in the front-end, your `next.config` will need to be configured. We will also use the `dotenv` package to load our `.env` file:
+
+`npm install --save dotenv`
+
+```
+// next.config.js
+require("dotenv").config();
+
+module.exports = {
+  env: {
+    GITHUB_CLIENT_ID: process.env.GITHUB_CLIENT_ID,
+    REPO_FULL_NAME: process.env.REPO_FULL_NAME,
+    BASE_BRANCH: process.env.BASE_BRANCH,
+  },
+  // ...
+}
+```
+
+For more help setting up environment variables with Next, see the [Next docs](https://nextjs.org/docs/api-reference/next.config.js/environment-variables)
+
+> Note that we did not add `GITHUB_CLIENT_SECRET`. This is the a secert key that should only be used from the server and should not be accessible through the browser.
+
+## Setup
+
+### Configure the _\_app.tsx_
+
+A few steps must be followed in order to edit your GitHub content using TinaCMS.
+
+1. **Create the TinaCMS**
+1. **Register the GithubClient:** The client accepts a string ('/api/proxy-github' in our case). All requests using the `GithubClient` gets passed through a proxy on our site. This allows us to securely attach the authentication tokens on the backend.
+1. Make sure the Sidebar is hidden unless we're in Next's [Preview/Edit mode](https://nextjs.org/docs/advanced-features/preview-mode).
+1. **Wrap the Page with `TinacmsGithubProvider`:** This component lets us authenticate with GitHub. It is given config and callbacks that hit our `/api` server functions to enable Preview/Edit Mode after authentication is complete.
+1. **Add a button for entering Preview/Edit Mode:** We must provide a means of triggering authentication. This a simple example of how to dow so.
+
+**pages/\_app.tsx**
```

Comment:
```suggestion
```

Reply:
To keep the style consistent, we've been adding the example file-paths as a comment in the code block